azurerm/internal/services/network/resource_arm_express_route_gateway.go

```diff
@@ -0,0 +1,203 @@
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for present of existing ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if resp.ID != nil && *resp.ID != "" {
+			return tf.ImportAsExistsError("azurerm_express_route_gateway", *resp.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	virtualHubId := d.Get("virtual_hub_id").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := network.ExpressRouteGateway{
+		Location: utils.String(location),
+		ExpressRouteGatewayProperties: &network.ExpressRouteGatewayProperties{
+			AutoScaleConfiguration: &network.ExpressRouteGatewayPropertiesAutoScaleConfiguration{
+				Bounds: expandExpressRouteGatewayAutoScaleConfigurationBounds(d),
+			},
+			VirtualHub: &network.VirtualHubID{
+				ID: &virtualHubId,
+			},
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read ExpressRoute Gateway %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmExpressRouteGatewayRead(d, meta)
+}
+
+func resourceArmExpressRouteGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteGatewaysClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["expressRouteGateways"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] ExpressRoute Gateway %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.ExpressRouteGatewayProperties; props != nil {
+		d.Set("virtual_hub_id", props.VirtualHub.ID)
+
+		if setErr := d.Set("scale_units", props.AutoScaleConfiguration.Bounds.Min); setErr != nil {
```

Comment:
there's a few things here:

1) there's a crash here if `props.AutoScaleConfiguration` or `props.AutoScaleConfiguration.Bounds` is nil
2) since this is a `*int32` there's no guarantees this'll be successfully handled by `d.Set` - so this behaviour may change in the future
3) since this is a simple type we don't need to handle the error here

as such could we make this:

```
scaleUnits := 0
if props.AutoScaleConfiguration != nil && props.AutoScaleConfiguration.Bounds != nil && props.AutoScaleConfiguration.Bounds.Min != nil {
  scaleUnits = *props.AutoScaleConfiguration.Bounds.Min
}
d.Set("scale_units", scaleUnits)
```

Reply:
Have made the suggested changes, with a minor alteration, let me know if there's an issue here.